widget/widget.go

```diff
@@ -12,6 +12,7 @@ type BaseWidget struct {
 	size     fyne.Size
 	position fyne.Position
 	Hidden   bool
+	cursor	 fyne.Cursor
```

Comment:
I don't think that every widget should implement `fyne.Cursor`. Only the entries and links should, for now. Every non-cursorable object uses the default cursor. This saves a lot of redundant tests.